lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -139,6 +147,31 @@ def get_request_id(headers, env)
           headers ||= {}
           headers['X-Request-Id'] || headers['X-Request-ID'] || env['HTTP_X_REQUEST_ID']
         end
+
+        private
+
+        REQUEST_SPAN_DEPRECATION_WARNING = %(
+          Use of :datadog_rack_request_span in the Rack env is DEPRECATED.
+          If you need the Rack request span, try using `Datadog.tracer.active_span`.
+          This key will no longer be available in future versions.).freeze
+
+        def add_deprecation_warnings(env)
+          env.instance_eval do
+            def [](key)
+              if key == :datadog_rack_request_span
+                Datadog::Tracer.log.warn(REQUEST_SPAN_DEPRECATION_WARNING)
```

Comment:
This one could be very spammy. If that internal attribute is used in a middleware, it means 1 log for each request.

Reply:
Adding an instance variable to scope this to once per request. Our own code shouldn't ever generate these warnings.